onnxruntime/core/session/IOBinding.cc

```diff
@@ -78,7 +78,16 @@ common::Status IOBinding::SynchronizeOutputs() {
   return Status::OK();
 }
 
-common::Status IOBinding::BindOutput(const std::string& name, const OrtValue& ml_value, OrtDevice device) {
+common::Status IOBinding::BindOutput(const std::string& name, const OrtValue& ml_value) {
```

Comment:
minor refactoring as suggested by @skottmckay  - split BindOutput() into 2 - one with an OrtValue (pre-allocated memory) and one with OrtDevice (no pre-allocation + telling Ort where to allocate). Removes all confusions with having both parameters together. 

CC: @yuslepukhin as this may be important info when working on C/C# APIs for IOBinding